python/lsst/ts/Dome/mock_llc_statuses/amcs_status.py

```diff
@@ -53,28 +55,48 @@ async def determine_status(self, current_tai):
-                        self.status = LlcStatus.STOPPED.value
-                    else:
-                        self.status = LlcStatus.PARKED.value
-            else:
-                self.position_actual = self.position_orig - azimuth_step
-                if self.position_actual <= self.position_cmd:
-                    self.position_actual = self.position_cmd
-                    self.position_orig = self.position_actual
-                    if self.status == LlcStatus.MOVING.value:
-                        self.status = LlcStatus.STOPPED.value
+            self.position_actual = self.position_orig + azimuth_step
+
+            # perform boundary checks for the current motion or crawl
+            if (
+                self.motion_velocity >= 0 and self.position_actual >= self.position_cmd
+            ) or (
+                self.motion_velocity < 0 and self.position_actual <= self.position_cmd
+            ):
+                self.position_actual = self.position_cmd
+                if self.status == LlcStatus.MOVING:
+                    if self.crawl_velocity > 0:
+                        # make sure that the dome never stops moving because it is crawling
+                        self.position_cmd = math.inf
+                        self.motion_velocity = self.crawl_velocity
+                        self.status = LlcStatus.CRAWLING
+                    elif self.crawl_velocity < 0:
+                        # make sure that the dome never stops moving because it is crawling
+                        self.position_cmd = -math.inf
+                        self.motion_velocity = self.crawl_velocity
+                        self.status = LlcStatus.CRAWLING
                     else:
-                        self.status = LlcStatus.PARKED.value
+                        # make sure that the dome has stopped moving because the requested crawl velocity is 0
+                        self.position_cmd = self.position_actual
+                        self.motion_velocity = self.crawl_velocity
+                        self.status = LlcStatus.STOPPED
+                elif self.status == LlcStatus.PARKING:
+                    self.position_cmd = self.position_actual
+                    self.motion_velocity = 0
+                    self.status = LlcStatus.PARKED
+                elif self.status == LlcStatus.CRAWLING:
+                    # this situation should never happen and probably never will because the commanded
+                    # position should have been set to +/- math.inf
+                    raise ValueError(
+                        f"Went beyond the limit of {self.position_cmd} while in status {self.status.value}. "
+                        f"This should not happen."
+                    )
+                else:
+                    raise ValueError(f"Unknown state {self.status.value}")
         self.llc_status = {
-            "status": self.status,
+            "status": self.status.value,
```

Comment:
FYI: if you use IntEnum (I'm being too lazy to check, but consider it if you aren't) then I think you can skip the `.value`: it should "just work".

Reply:
The statuses sent by the LLCs contain strings like "Moving" and "Parked" and not "MOVING", "PARKED" or 3. So IntEnum will not work here.

Reply:
That is a fair point, though in fact enum values can be TitleCase (and if you only want the strings and not associated integers then IntEnum is irrelevant).